src/utils/image/index.ts

```diff
@@ -12,21 +12,49 @@ const compressAndPrepareFormData = async (imagePath: string) => {
 };
 
 const uploadImage = async (imagePath: string) => {
-  const formData = await compressAndPrepareFormData(imagePath);
+  try {
+    const formData = await compressAndPrepareFormData(imagePath);
+
+    const imageUrl = await uploadPhoto(formData);
+    return imageUrl;
+  } catch (error) {
+    console.log(error);
+    return imagePath;
+  }
+};
 
-  const imageUrl = await uploadPhoto(formData);
-  return imageUrl;
+const uploadFile = async (uri: string, name: string, type: string) => {
+  try {
+    const formData = new FormData();
+    formData.append('photo', {
+      uri,
+      name,
+      type
+    });
+
+    const url = await uploadPhoto(formData);
+    return url;
+  } catch (error) {
+    console.log(error?.response);
+    return uri;
+  }
 };
 
 const uploadImageWithoutAuth = async (imagePath: string) => {
-  const formData = await compressAndPrepareFormData(imagePath);
-
-  const imageUrl = await uploadPhotoWithoutAuth(formData);
-  return imageUrl;
+  try {
+    const formData = await compressAndPrepareFormData(imagePath);
+
+    const imageUrl = await uploadPhotoWithoutAuth(formData);
+    return imageUrl;
+  } catch (error) {
+    console.log(error);
+    return imagePath;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `uploadImageWithoutAuth` function, similar to the previous two functions, it would be better to throw an error or return a specific error message rather than returning the original image path.

```diff
-    console.log(error);
-    return imagePath;
+    throw new Error(`Failed to upload image without auth: ${error.message}`);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->